### PR TITLE
Initialize the Japanese Localization Guidelines

### PR DESCRIPTION
**Pull request by Okabe-Junya**
ref. #153 
Deploy preview: [link](https://github.com/cncf/techdocs/blob/115d2988e95b9da81893121c697d62fb0fdbbdad/docs/localization/ja/README.md)

## Feature Description

- Initialize the Japanese Localization Guidelines
    - for new contributors and many members

The Japanese Localization Guidelines include the following contents:

- Style Guide
- How to write specific terms in Japanese
- List of projects with ongoing Japanese localization, and the leads and l10n reviewers for each project
- Communication methods

## Background

https://github.com/cncf/techdocs/issues/153#issuecomment-2123992860

> Our project started with the cncf/glossary, and now we are working on l10n for other TAGs and WGs (for example, tag-runtime#145, tag-app-delivery#521). Through this, we have realized the importance of more general l10n documentation (and we think this issue is the best place to discuss it! Thank you @/nate-double-u ).

### CODE REVIEW

**Comment by Okabe-Junya** [comment]
To move forward this PR, I would like to do the following:

1. Accept reviews from Japanese localization reviewers for about a week (until 2024-06-23 00:00 UTC)
2. Have the reviewers from cncf/techdocs review and merge

File: docs/localization/ja/README.md
Comment (Okabe-Junya) [inline docs/localization/ja/README.md:59]: We need one more reviewer. How about you, @kenta-iijima??

File: docs/localization/ja/README.md
Comment (Okabe-Junya) [inline docs/localization/ja/README.md:67]: @naonishijima @kaitoii11 @hhiroshell

Since you are actively contributing to each project, I have listed your name as the lead/reviewer. Is that okay?

Comment (naonishijima) [reply docs/localization/ja/README.md:67]: Okay!

File: docs/localization/ja/README.md
Comment (Okabe-Junya) [reply docs/localization/ja/README.md]: Thank you :)

https://github.com/cncf/techdocs/pull/259/commits/e68faa5d1e98f1146f07866035a87716a1954cf8

**Review by kaitoii11 — APPROVED** [review]

**Comment by Okabe-Junya** [comment]
Thank you, @nate-double-u!!

> Could you update this PR to add a section to the CNCF Localization Guides page (/docs/localization/README.md) to introduce these language-specific guides?

Ok!! I added `## Localization READMEs` section, https://github.com/cncf/techdocs/pull/259/commits/bb84976125b46934604070bf0be76448f275bcee. 

I think it would be good to have an English version of this as well (even if it's very rough or just an outline)

SGTM!! Now, I created localization template and will open another PR - IMO, it would be better to create another PR. Thank you :)

**Comment by nate-double-u** [comment]
> Thank you, @nate-double-u!!

😊

> > Could you update this PR to add a section to the CNCF Localization Guides page (/docs/localization/README.md) to introduce these language-specific guides?
> 
> Ok!! I added `## Localization READMEs` section, [bb84976](https://github.com/cncf/techdocs/commit/bb84976125b46934604070bf0be76448f275bcee).

Perfect, thanks!

> > I think it would be good to have an English version of this as well (even if it's very rough or just an outline)
> 
> SGTM!! Now, I created localization template and will open another PR - IMO, it would be better to create another PR. Thank you :)

That's a good idea, it is a separate (related) thing, and should be in a separate PR—good call.

**Comment by nate-double-u** [comment]
I think that running `npm run fix:format` locally will fix the format/link checks.

**Comment by Okabe-Junya** [comment]
Thank you :)

> I think that running npm run fix:format locally will fix the format/link checks.

TBH, prettier does not work well for Japanese, since prettier breaks line after a certain number of characters (Please not that Japanese doesn't break lines except "。" (like `.`))

So, I want to disable linter (at least the Japanese README)

**Comment by nate-double-u** [comment]
That's fair—I think you can disable things per folder. @chalin to confirm as the lead on spearheading the reformat project (note, they are away this week)

File: docs/localization/ja/README.md
Comment (hhiroshell) [reply docs/localization/ja/README.md]: 👍

**Comment by Okabe-Junya** [comment]
I’ve disabled prettier for Japanese README, https://github.com/cncf/techdocs/pull/259/commits/e2e6bf10fbda711801d0e6fceb5e3f0f0fe2695a 

(I'll revert it if there are any issues.)

**Comment by chalin** [comment]
Hi all. Thank you for this initiative @Okabe-Junya. I was away. I'm back but still catching up. I'll get to this as soon as I can. In the meantime, let me cc a few extra people for feedback:

- @a-mccarthy
- @open-telemetry/docs-ja-approvers: @katzchang, @ymotongpoo

**Comment by Okabe-Junya** [comment]
Thank you, I appreciate your support, chalin!

I have already informed katzchang, ymotongpoo about this PR. (I announced this PR on the [`#glossary-localization-japanese`](https://cloud-native.slack.com/archives/C057F81GFUG/p1718467890051979?thread_ts=1717787015.407889&cid=C057F81GFUG) channel in CNCF Slack, and they are there!)

Once we receive feedback from a-mccarthy, we can move forward! Thank you :)

**Comment by chalin** [comment]
Thank you for the clarifications @Okabe-Junya.

In addition to @a-mccarthy's feedback, I would appreciate if the following could **formally approve** this PR:

- All the currently listed reviewers on this PR
- @katzchang
- @ymotongpoo

**Review by naonishijima — APPROVED** [review]
LGTM

**Review by hhiroshell — APPROVED** [review]
Thanks for your hard work!

**Comment by ymotongpoo** [comment]
LGTM

**Review by ymotongpoo — APPROVED** [review]

**Comment by katzchang** [comment]
LGTM👍

**Review by katzchang — APPROVED** [review]

File: .prettierignore
Comment (chalin) [inline .prettierignore:2]:
```suggestion
/docs/localization/ja
```

File: docs/localization/ja/README.md
Comment (chalin) [inline docs/localization/ja/README.md:20]: Consider instead writing that the project's style guide should be followed, using the K8s style guide as a fall back where there is none (or something to that effect).

File: docs/localization/ja/README.md
Comment (chalin) [inline docs/localization/ja/README.md:6]: You write:

> This may not be applicable to some projects. Please consult with the project maintainers, community, and Japanese localization lead as appropriate.

A few comments:

- Can you be more specific and write "This guide may not ...", that is:
  このガイドは一部のプロジェクトには適用できない場合があります。
- How would the reader know how to contact the "Japanese localization lead"? At a minimum, link to the コミュニケーション section.
- How about this for the second sentence or did you really want both the project maintainers _and_ the Japanese localization lead to be contacted? I'd drop the mention of "community".
  必要に応じて、プロジェクト管理者または日本語ローカリゼーション担当者にご相談ください。

File: docs/localization/ja/README.md
Comment (chalin) [inline docs/localization/ja/README.md:67]: I think that the list of leads and reviewers should be provided by the project, and the project only. This table should refer to the specific page and section in the project repo where the localization lead and reviewers are listed. 

This will help us keep things [DRY](https://ja.wikipedia.org/wiki/Don%27t_repeat_yourself).

**Review by chalin — COMMENTED** [review]
Thank you @Okabe-Junya. See inline comments.

File: docs/localization/ja/README.md
Comment (Okabe-Junya) [reply docs/localization/ja/README.md:67]: This is a very good point. However, the maturity of localization varies from project to project, and many projects do not manage who the localization leads/reviewers are.

A major reason we are creating this guide is to provide a single entry point for new contributors to refer to at the beginning. So, I think this list is very important.

Of course, our goal is for reviewers to be managed by each project as well, but that will take some time. We look forward to that happening in the future :)

File: docs/localization/ja/README.md
Comment (Okabe-Junya) [reply docs/localization/ja/README.md]: Thank you!
> * Can you be more specific and write "This guide may not ...", that is:
>   このガイドは一部のプロジェクトには適用できない場合があります。

I revised the sentence to "How much this guideline is applied to each project is determined by the localization lead of each project"

> * How would the reader know how to contact the "Japanese localization lead"? At a minimum, link to the コミュニケーション section.

I think it means mentioning on GitHub, so I think a table of projects is necessary.

> * How about this for the second sentence or did you really want both the project maintainers _and_ the Japanese localization lead to be contacted? I'd drop the mention of "community".
>   必要に応じて、プロジェクト管理者または日本語ローカリゼーション担当者にご相談ください。

Good point! fixed  and revised it!

File: docs/localization/ja/README.md
Comment (Okabe-Junya) [reply docs/localization/ja/README.md:20]: That is a very good point. However, the Kubernetes localization is very mature, and this localization guide is excellent. Also, the "Style Guide" section contains only general rules, so I would like to start by referring to it.

Additionally, I have been leading most of the recent updates to this document. 

FYI: https://github.com/kubernetes/website/commits/main/content/ja/docs/contribute/localization.md?since=2023-01-01&until=2024-06-28

I do not want to update two similar style guides (as you mentioned, this is a DRY practice).

**Comment by Okabe-Junya** [comment]
Thank you for the wonderful review, @chalin. Your feedback is very thorough and helpful!

Based on your comments, I have made some revisions. I have also replied to some of your comments (to discuss)

Thanks!

File: docs/localization/ja/README.md
Comment (chalin) [reply docs/localization/ja/README.md:20]: Ok, if that is your preference. 👍🏻

File: docs/localization/ja/README.md
Comment (chalin) [reply docs/localization/ja/README.md:67]: Ok, we can leave this as is for now.

**Review by chalin — APPROVED** [review]
LGTM, thank you for the updates.

**Comment by Okabe-Junya** [comment]
Thank you for your support, chalin!

**Review by nate-double-u — APPROVED** [review]
Thanks @Okabe-Junya, and everyone on the Japanese localization team. Great work here!

**Pull request merged**
